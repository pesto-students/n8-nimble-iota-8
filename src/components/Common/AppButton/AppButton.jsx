import React from "react";
import { Button } from "antd";
import PropTypes from "prop-types";
import styles from "./Button.module.less";

const AppButton = ({ type, size, loading, onClick, children }) => {
    return (
        <React.Fragment>
            <Button
                type={type ? type : "primary"}
                size={size}
                loading={loading}
                className={styles.btn}
                onClick={onClick}
            >
                {children}
            </Button>
        </React.Fragment>
    );
};

export const FullLengthButton = ({
    type,
    size,
    loading,
    onClick,
    children,
}) => {
    return (
        <React.Fragment>
            <Button
                type={type ? type : "primary"}
                size={size}
                loading={loading}
                className={styles.fulllenbtn}
                onClick={onClick}
            >
                {children}
            </Button>
        </React.Fragment>
    );
};

export default AppButton;

AppButton.propTypes = {
    type: PropTypes.oneOf("primary", "dashed", "text", "link", "default"),
    size: PropTypes.oneOf("large", "middle", "small"),
    loading: PropTypes.bool,
    children: PropTypes.elementType,
    onClick: PropTypes.func,
};
FullLengthButton.propTypes = {
    type: PropTypes.oneOf("primary", "dashed", "text", "link", "default"),
    size: PropTypes.oneOf("large", "middle", "small"),
    loading: PropTypes.bool,
    children: PropTypes.elementType,
    onClick: PropTypes.func,
};
