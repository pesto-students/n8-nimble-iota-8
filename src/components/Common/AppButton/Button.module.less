.btn {
    border-radius: 0.5em;
    box-shadow: 0 0.25rem 0.75rem rgba(#3751ff, 0.25);
}
.fulllenbtn {
    border-radius: 0.5em;
    box-shadow: 0 0.25rem 0.75rem rgba(#3751ff, 0.25);
    width: 100%;
}
