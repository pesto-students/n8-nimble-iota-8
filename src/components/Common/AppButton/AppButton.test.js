import React from "react";
import renderer from "react-test-renderer";
import AppButton from "src/components/Common/AppButton/AppButton";
it("ActiveMark renders correctly", () => {
    const tree = renderer.create(<AppButton>Button</AppButton>).toJSON();
    expect(tree).toMatchSnapshot();
});
