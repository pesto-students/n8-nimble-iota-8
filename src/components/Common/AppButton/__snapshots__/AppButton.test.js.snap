// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveMark renders correctly 1`] = `
<button
  className="ant-btn ant-btn-primary"
  onClick={[Function]}
  type="button"
>
  <span>
    Button
  </span>
</button>
`;
